pennylane/devices/default_qubit.py

```diff
@@ -204,6 +232,45 @@ def define_wire_map(self, wires):
         wire_map = zip(wires, consecutive_wires)
         return dict(wire_map)
 
+    #def is_all_ops_supported(self):
+    #    ops_list =self.op_queue()
+    #    tmp = bool(True)
+    #    for index, operation in enumerate(ops_list):
+    #        if operation not in self.operations:
+    #            return bool(False)
+    #        else:
+    #            tmp*=bool(True)
+    #    return tmp 
+
+    @staticmethod
+    def supports_operation_backward(self, operation):
```

Comment:
While currently `supports_operation` and `supports_observable` take either a type or a string, that is an extremely problematic call signature that I'm trying to get rid of, as it fails to account for instance-specific parameters and keywords.

For example,  `Pow(op, 2)` is supported on both the forward and backward backpropagation pass, but `Pow(op, 1.5)` is only supported on the forward backpropgation pass. We use use scipy's fractional matrix power, since the ML frameworks don't have fractional matrix powers defined.